docs/pubsub-api.rst

```diff
@@ -0,0 +1,255 @@
+   >>> subscription = topic.create_subscription('subscription_name',
+   ...                                          push_endpoint=ENDPOINT)
+
+Get an extant subscription for a topic:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> subscription = topic.get_subscription('subscription_name')
+
+Attempt to get a non-extant subscription for a topic:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> subscription = topic.get_subscription('nonesuch')
+   Traceback (most recent call last):
+     File "<stdin>", line 1, in ?
+   NotFound: ...
+
+Update the ACK deadline for a subscription:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> subscription = topic.get_subscription('subscription_name')
+   >>> subscription.modify_ack_deadline(90)
+
+Convert a "pull" subscription to "push":
+
+.. doctest::
+
+   >>> ENDPOINT = 'https://example.com/hook'
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> subscription = topic.get_subscription('subscription_name')
+   >>> subscription.modify_push_configuration(push_endpoint=ENDPOINT)
+
+Convert a "push" subscription to "pull":
+
+.. doctest::
+
+   >>> ENDPOINT = 'https://example.com/hook'
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> subscription = topic.create_subscription('subscription_name',
+   ...                                          push_endpoint=ENDPOINT)
+   >>> subscription.modify_push_configuration(push_endpoint=None)
+
+List extant subscriptions for a topic:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> [subscription.name for subscription in topic.list_subscriptions()]
+   ['subscription_name']
+
+Delete a subscription:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> subscription = topic.get_subscription('subscription_name')
+   >>> subscription.delete()
+
+
+Pull messages from a subscription
+---------------------------------
+
+Fetch pending messages for a "pull" subscription (the messages will have
+been ACKed already):
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> subscription = topic.get_subscription('subscription_name')
+   >>> [message.id for message in subscription.pull()]
+   [<message_id1>, <message_id2>, ...]
+
+Fetch a limited number of pending messages for a "pull" subscription:
+
+.. doctest::
+
+   >>> from gcloud import pubsub
+   >>> topic = pubsub.get_topic('topic_name')
+   >>> subscription = topic.get_subscription('subscription_name')
+   >>> [message.id for message in subscription.pull(max_messages=2)]
+   [<message_id1>, <message_id2>]
+
+Fetch messages for a "pull" subscription without blocking (none pending):
```

Comment:
What is the underlying behavior here? How does it compare to explicitly specifying a `max_messages` value?